R/basic.R

```diff
@@ -36,7 +36,12 @@
 #' is_igraph(g)
 #' is_igraph(numeric(10))
 is_igraph <- function(graph) {
-  "igraph" %in% class(graph)
+  if (!inherits(graph, "igraph")) {
+    return(FALSE)
+  }
+
+  warn_version(graph)
+  TRUE
```

Comment:
`is_igraph()` is called by `ensure_igraph()`, which in turn is called by (almost?) every public function.

So this way we get a version check in every function.

But if there's a version mismatch, the message will say:

> This graph was created by an old(er) igraph version.
> Call upgrade_graph() on it to use with the current igraph version
> **For now we convert it on the fly...**

Is the statement on the last line (bolded) accurate?

Reply:
I've added tests in https://github.com/igraph/rigraph/pull/832/files#diff-58f022fa113e96e607f6d9f3816676cd2350b79cbc4546a8861543925cc670b3, but this is out of scope for this PR. These tests should trigger when we bump the version to 1.5.0 in #835.